### PR TITLE
update appium service

### LINKED ISSUE #4081: Appium service: move appium to dev dependency and update cli
**Is your feature request related to a problem? Please describe.**
Appium service has `appium` as a dependency. This is a restriction for users who want to use appium different from the one is shipped.

**Describe the solution you'd like**
- move appium to dev dependency 
- also install appium same as chromedriver on initial setup if it is not installed globally

### PR DESCRIPTION
**Pull request by mgrybyk**
## Proposed changes

Remove `appium` from `@wdio/appium-service` dependencies and install `appium` same as `chromedriver` if it is not installed globally.

fixed #4081 

also
- use outputDir as folder for appium logs if it is not specified in appium
- fixed `chromedriver` condition in `wdio.conf.tpl.ejs`
- `chromedriver` and `appium` packages are added when using `wdio install service ...`
- fixed some failing tests because of Date.now()

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

tested changes in my project.

### CODE REVIEW

File: packages/wdio-appium-service/package.json [hunk 1]
```diff
@@ -33,8 +33,10 @@
   "bugs": {
     "url": "https://github.com/webdriverio/webdriverio/issues"
   },
+  "devDependencies": {
+    "appium": "^1.13.0"
```
Comment (christian-bromann) [inline packages/wdio-appium-service/package.json]: I am not sure if we need Appium whatsoever because in dev I don't really need it too

**Comment by mgrybyk** [comment]
> If neither appium is installed globally nor as dev dependency, can we install it for the user?

https://github.com/webdriverio/webdriverio/pull/4089/files#diff-67c78548af6da581e175d98914982e23R42

File: packages/wdio-appium-service/package.json (hunk 1 above)
Comment (mgrybyk) [reply packages/wdio-appium-service/package.json]: let me check and remove if possible

Comment (mgrybyk) [reply packages/wdio-appium-service/package.json]: there were some changes required in test but I've removed appium.

Comment (mgrybyk) [reply packages/wdio-appium-service/package.json]: there were some changes required in test but I've removed appium.

**Comment by christian-bromann** [comment]
> it is done same as for chromedriver, we are installing appium if it is not installed globally

Yeah but this applies only if people to the setup wizard. What happens if they add it to their existing framework.

**Comment by mgrybyk** [comment]
Got your point, I'll check. Completely forgot about that, thank you

**Comment by mgrybyk** [comment]
@christian-bromann if people add either appium or chromedriver service no additional packages will be installed right now

**Comment by christian-bromann** [comment]
@mgrybyk with this change for the appium service, yes. That is why I would suggest to install appium or download chromedriver for these cases, or have a clear error message saying what is needed to do.

**Comment by mgrybyk** [comment]
ok, I'll change behavior for appium and chromedriver

**Comment by mgrybyk** [comment]
@baruchvlz you might be interested as well.

**Comment by mgrybyk** [comment]
@christian-bromann 
`./bin/wdio.js install service appium` installs also `appium` if it is not globally installed, otherwise prints message saying that it is needed to add `appium: { command: 'appium' }`

`./bin/wdio.js install service chromedriver` installs also `chromedriver`. It should mention that `path` have to be changes to `/` -> `path: '/'`

File: packages/wdio-appium-service/src/launcher.js [hunk 2]
```diff
@@ -69,8 +69,16 @@ export class AppiumLauncher {
         this.process.stderr.pipe(logStream)
     }
 
-    _getAppiumCommand() {
-        return require.resolve('appium')
+    _getAppiumCommand(moduleName = 'appium') {
+        try {
+            return require.resolve(moduleName)
+        } catch (err) {
+            log.error('appium is not installed locally.\n' +
```
Comment (christian-bromann) [inline packages/wdio-appium-service/src/launcher.js]:
```suggestion
            log.error('appium is not installed locally. Please install Appium via `npm install appium`.\n' +
```

File: packages/wdio-appium-service/src/launcher.js [hunk 3]
```diff
@@ -69,8 +69,16 @@ export class AppiumLauncher {
...
+            'If you use globally installed appium please add\n' +
```
Comment (christian-bromann) [inline packages/wdio-appium-service/src/launcher.js]:
```suggestion
            'If you use globally installed Appium please add\n' +
```

**Pull request merged**